lib/rules/disallow-function-prototype-extension.js

```diff
@@ -0,0 +1,36 @@
+"use strict";
+
+var includes = require("../utils/includes");
+var extensionFixes = {
+  'property': "Use the computed('dependentKey', function() {}) syntax instead.",
+  'observes': "Use the observer('dependentKey', function() {}) syntax instead.",
+  'observesBefore': "Also, using before observers is deprecated."
+}
+var prohibitedFunctionMethods = Object.keys(extensionFixes);
+
+module.exports = function(context) {
+  return {
+    'CallExpression': function(node) {
+      if (
+        node.callee.object && (
+          node.callee.object.type === 'FunctionExpression' ||
+          node.callee.object.type === 'CallExpression'
+        ) &&
+        includes(prohibitedFunctionMethods, node.callee.property.name)) {
```

Comment:
Seems odd to make an array and use `includes` when you could just look up the name on the `extensionFixes` object?